pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -294,16 +283,24 @@ func (c *Cluster) UpdateActiveMgrLabel(daemonNameToUpdate string, prevActiveMgr
 			if err != nil {
 				// In case of failure we report as 'active manager' the previous value, this way
 				// we force refreshing mgrRoleLabelName label next time this function is called
-				currActiveMgr = prevActiveMgr
+				allPodsUpdated = false
 				logger.Warningf("cannot update the active mgr pod %q. err=%v", pods.Items[i].Name, err)
 			}
 		}
 	}
 
-	return currActiveMgr, c.reconcileServices()
+	err = c.reconcileServices()
+	if err != nil {
+		return fmt.Errorf("unable to reconcile services: %w", err)
+	}
+	if !allPodsUpdated {
+		return errors.New("not all pod labels were updated: reconcile again")
```

Comment:
If we keep the last error as suggested above, then we can return it. Otherwise, the caller doesn't know the details of the failure.
```suggestion
		return podUpdateErr
```